File: src/main/java/no/mesan/fag/patterns/timesheet/external/TimeSource.java
package no.mesan.fag.patterns.timesheet.external;

import no.mesan.fag.patterns.timesheet.data.TimesheetEntry;

import java.io.File;
import java.io.IOException;
import java.nio.charset.Charset;
import java.nio.file.Files;
import java.util.Arrays;
import java.util.Iterator;
import java.util.LinkedList;
import java.util.List;

/** Kilde til timedata. */
public class TimeSource implements Iterable<TimesheetEntry> {

    private static List<String> readAllLines(final String fileName) {
        final List<String> res = new LinkedList<>();
        try {
            final List<String> lines = Files.readAllLines(new File(fileName).toPath(), Charset.defaultCharset());
            for (final String line : lines) {
                res.addAll(Arrays.asList(line.split(" +")));
            }
            return res;
        }
        catch (IOException e) {
            e.printStackTrace();
            return res;
        }
    }

    private static final List<TimesheetEntry> ENTRIES=
            TimesheetEntry.create(readAllLines("src/main/resources/timer.txt"));

    @Override
    public Iterator<TimesheetEntry> iterator() {
        return ENTRIES.iterator();
    }
}

File: src/main/scala/no/mesan/fag/patterns/scala/timesheet/external/TimeSource.scala
package no.mesan.fag.patterns.scala.timesheet.external

import no.mesan.fag.patterns.scala.timesheet.data.TimesheetEntry
import scala.io.Source

/** Kilde til timedata. */
object TimeSource extends Iterable[TimesheetEntry] {

  private def getContents(fileName:String): List[String] = {
    val source = Source.fromFile(fileName, "ISO-8859-1")
    for (l <- source.getLines().toList; s<- l.split(" +")) yield s
  }

  val Entries = TimesheetEntry(getContents("src/main/resources/timer.txt"))
  override def iterator: Iterator[TimesheetEntry] =  Entries.iterator
}
